File: repos_estudiantes_ALVAREZ_tercera_entrega/CANAVIRI_ALTAMIRANO_AYLIN_MICHEL/PERSISTENCIADEOBJETOS/EJERCICIO5/Ej5-PYTHON/Farmacia.py
from Medicamento import Medicamento

class Farmacia:
    def __init__(self, nombreFarmacia="", sucursal=0, direccion=""):
        self.__nombreFarmacia = nombreFarmacia
        self.__sucursal = sucursal
        self.__direccion = direccion
        self.__nroMedicamentos = 0
        self.__m = []

    def leer(self):
        self.__nombreFarmacia = input("Nombre de la farmacia: ")
        self.__sucursal = int(input("Número de sucursal: "))
        self.__direccion = input("Dirección: ")
        n = int(input("Cantidad de medicamentos: "))
        for _ in range(n):
            med = Medicamento()
            med.leer()
            self.adicionarMedicamento(med)

    def mostrar(self):
        print(f"Farmacia: {self.__nombreFarmacia}, Sucursal: {self.__sucursal}, Dirección: {self.__direccion}")
        if self.__nroMedicamentos == 0:
            print("  Sin medicamentos.")
        for med in self.__m:
            med.mostrar()

    def getDireccion(self):
        return self.__direccion

    def getSucursal(self):
        return self.__sucursal

    def mostrarMedicamentos(self, tipo):
        for med in self.__m:
            if med.getTipo().lower() == tipo.lower():
                med.mostrar()

    def buscaMedicamento(self, nombreMed):
        for med in self.__m:
            if med.getNombre().lower() == nombreMed.lower():
                return True
        return False

    def adicionarMedicamento(self, medicamento):
        if self.__nroMedicamentos < 100:
            self.__m.append(medicamento)
            self.__nroMedicamentos += 1
        else:
            print("¡Límite de 100 medicamentos alcanzado!")

    def getListaMedicamentos(self):
        return self.__m

File: repos_estudiantes_TARQUI_tercera_entrega/ILLANES_VASQUEZ_GABRIEL/Practica 3 Aux/PERSISTENCIA/Ejercicio5/Farmacia.java
package Ejercicio5;

import java.util.Scanner;

public class Farmacia {
    private String nombreFarmacia;
    private int sucursal;
    private String direccion;
    private int nroMedicamentos;
    private Medicamento[] m = new Medicamento[100];

    public Farmacia() {
        for (int i = 0; i < m.length; i++)
            m[i] = new Medicamento();
    }

    public void leer() {
        Scanner sc = new Scanner(System.in);
        System.out.print("Nombre Farmacia: ");
        nombreFarmacia = sc.nextLine();
        System.out.print("Número de sucursal: ");
        sucursal = sc.nextInt(); sc.nextLine();
        System.out.print("Dirección: ");
        direccion = sc.nextLine();
        System.out.print("Cantidad de medicamentos: ");
        nroMedicamentos = sc.nextInt();
        for (int i = 0; i < nroMedicamentos; i++) {
            System.out.println("Medicamento " + (i + 1));
            m[i].leer();
        }
    }

    public void mostrar() {
        System.out.println("Farmacia: " + nombreFarmacia + " | Sucursal: " + sucursal + " | Dirección: " + direccion);
        for (int i = 0; i < nroMedicamentos; i++) {
            m[i].mostrar();
        }
    }

    public String getDireccion() {
        return direccion;
    }

    public int getSucursal() {
        return sucursal;
    }

    public void mostrarMedicamentos(int x) {
        if (sucursal == x) {
            for (int i = 0; i < nroMedicamentos; i++) {
                if (m[i].getTipo().equalsIgnoreCase("Tos")) {
                    m[i].mostrar();
                }
            }
        }
    }

    public boolean buscaMedicamento(String n) {
        for (int i = 0; i < nroMedicamentos; i++) {
            if (m[i].getNombre().equalsIgnoreCase(n)) {
                return true;
            }
        }
        return false;
    }

    public String getDatos() {
        StringBuilder sb = new StringBuilder();
        sb.append(sucursal).append(";").append(direccion).append("\n");
        for (int i = 0; i < nroMedicamentos; i++) {
            sb.append(m[i].getNombre()).append(",")
              .append(m[i].getTipo()).append(",")
              .append(m[i].getPrecio()).append("\n");
        }
        sb.append("###\n");
        return sb.toString();
    }
}
